sdk/core/src/main/java/com/klaviyo/coresdk/Klaviyo.kt

```diff
@@ -1,73 +1,143 @@
-     * Additional properties associated to the event that are not for identifying the customer
+     * @param email Email address for active user
+     * @return
      */
-    fun track(
-        event: KlaviyoEvent,
-        customerProperties: KlaviyoCustomerProperties,
-        properties: KlaviyoEventProperties? = null
-    ) {
-        val request = TrackRequest(KlaviyoConfig.apiKey, event, customerProperties, properties)
-        processRequest(request)
-    }
+    fun setEmail(email: String): Klaviyo = this.setProfile(Profile().setEmail(email))
+
+    /**
+     * Assigns a phone number to the current internally tracked profile
+     *
+     * The SDK keeps track of current profile details to
+     * build analytics requests with profile identifiers
+     *
+     * This should be called whenever the active user in your app changes
+     * (e.g. after a fresh login)
+     *
+     * @param phoneNumber Phone number for active user
+     */
+    fun setPhoneNumber(phoneNumber: String): Klaviyo =
+        this.setProfile(Profile().setPhoneNumber(phoneNumber))
+
+    /**
+     * Assigns an external ID to the current internally tracked profile
+     *
+     * The SDK keeps track of current profile details to
+     * build analytics requests with profile identifiers
+     *
+     * This should be called whenever the active user in your app changes
+     * (e.g. after a fresh login)
+     *
+     * @param id Phone number for active user
+     * @return
+     */
+    fun setExternalId(id: String): Klaviyo = this.setProfile(Profile().setIdentifier(id))
+
+    /**
+     * Assign arbitrary attributes to the current profile by key
+     *
+     * This should be called when you collect additional data about your user
+     * (e.g. first and last name, or an address)
+     *
+     * @param propertyKey
+     * @param value
+     * @return
+     */
+    fun setProfileAttribute(propertyKey: KlaviyoProfileAttributeKey, value: String): Klaviyo =
+        setProfile(Profile().also { it[propertyKey] = value })
 
     /**
-     * Creates an identify request for the Klaviyo APIs and begins processing.
+     * Debounce timer for enqueuing profile API calls
+     */
+    private var timer: Clock.Cancellable? = null
+
+    /**
+     * Queues a request to identify profile properties to the Klaviyo API
+     *
+     * Any new identifying properties (externalId, email, phone) will be saved to the current
+     * internally tracked profile. Otherwise any identifiers missing from [profile] will be
+     * populated from the current internally tracked profile info.
+     *
      * Identify requests track specific properties about a user without triggering an event
      *
-     * @param properties A map of properties that define the user
+     * @param profile A map of properties that define the user
+     * @return
+     */
+    fun setProfile(profile: Profile): Klaviyo = apply {
+        UserInfo.mergeProfile(profile)
```

Comment:
Oh crap this is definitely not right, its going to lose all but the most recent function call's profile attributes. 
It needs a pending profile object to merge into as multiple calls are made.

Reply:
Fixed and added tests of the resultant API call from the debouncing